### PR TITLE
feat: add SsmClient bean

### PR DESCRIPTION
**Pull request by ttzn**
Having a bean to query the Parameter Store sounds useful. This should be a pretty painless MR. I'm marking this as WIP as I need to make a couple more checks but I'd like your opinion.

### CODE REVIEW

**Comment by graemerocher** [comment]
Build is actually passing, we need to fix the test annotation thing

File: src/main/docs/guide/sdkv2/ssm.adoc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+To use an SSM client, add the following dependency:
+
+dependency:sqs[groupId="software.amazon.awssdk"]
```
Comment (alvarosanchez) [inline src/main/docs/guide/sdkv2/ssm.adoc:3]:
```suggestion
dependency:ssm[groupId="software.amazon.awssdk"]
```

**Review by alvarosanchez — CHANGES_REQUESTED** [review]

**Review by alvarosanchez — APPROVED** [review]

File: aws-sdk-v2/src/test/groovy/io/micronaut/aws/sdk/v2/service/SsmClientSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package io.micronaut.aws.sdk.v2.service
+
+import io.micronaut.context.ApplicationContext
+import software.amazon.awssdk.services.ssm.SsmAsyncClient
+import software.amazon.awssdk.services.ssm.SsmClient
+import spock.lang.Specification
+
+class SsmClientSpec extends Specification{
```
Comment (sdelamo) [inline aws-sdk-v2/src/test/groovy/io/micronaut/aws/sdk/v2/service/SsmClientSpec.groovy:8]: please, add a trailing space. 

replace: `Specification{`

with `Specification {`

File: aws-sdk-v2/src/test/groovy/io/micronaut/aws/sdk/v2/service/SsmClientSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    void "it can configure a sync client"() {
+        given:
+        ApplicationContext applicationContext = ApplicationContext.run()
+
+        when:
+        SsmClient client = applicationContext.getBean(SsmClient)
+
+        then:
+        client.serviceName() == SsmClient.SERVICE_NAME
```
Comment (sdelamo) [inline aws-sdk-v2/src/test/groovy/io/micronaut/aws/sdk/v2/service/SsmClientSpec.groovy:18]: please, close the application context. Add: 

```groovy
cleanup: 
applicationContext.close()
```

File: aws-sdk-v2/src/test/groovy/io/micronaut/aws/sdk/v2/service/SsmClientSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    void "it can configure an async client"() {
+        given:
+        ApplicationContext applicationContext = ApplicationContext.run()
+
+        when:
+        SsmAsyncClient client = applicationContext.getBean(SsmAsyncClient)
+
+        then:
+        client.serviceName() == SsmClient.SERVICE_NAME
```
Comment (sdelamo) [inline aws-sdk-v2/src/test/groovy/io/micronaut/aws/sdk/v2/service/SsmClientSpec.groovy:29]: please, close the application context. add: 

```groovy
cleanup: 
applicationContext.close()
```

**Review by sdelamo — CHANGES_REQUESTED** [review]
Thanks for the contribution. Please, address the comments and I will merge the PR.

**Comment by ttzn** [comment]
@sdelamo I went all the way and improved all other tests in sdk-v2.

**Comment by sdelamo** [comment]
thanks @ttzn 

I've moved the application context close in tests to https://github.com/micronaut-projects/micronaut-aws/pull/808

**Comment by sdelamo** [comment]
Closed via: https://github.com/micronaut-projects/micronaut-aws/pull/810

**Pull request closed without merging**